meteor/server/api/showStyles.ts

```diff
@@ -75,50 +75,94 @@ export async function insertShowStyleBaseInner(organizationId: OrganizationId |
-): Promise<ShowStyleVariantId> {
+async function assertShowStyleBaseAccess(context: MethodContext | Credentials, showStyleBaseId: ShowStyleBaseId) {
 	check(showStyleBaseId, String)
 
 	const access = await ShowStyleContentWriteAccess.anyContent(context, showStyleBaseId)
 	const showStyleBase = access.showStyleBase
 	if (!showStyleBase) throw new Meteor.Error(404, `showStyleBase "${showStyleBaseId}" not found`)
-
-	return insertShowStyleVariantInner(showStyleBase, name)
 }
-export async function insertShowStyleVariantInner(
-	showStyleBase: ShowStyleBaseLight,
+
+export async function createDefaultShowStyleVariant(
+	context: MethodContext | Credentials,
+	showStyleBaseId: ShowStyleBaseId,
 	name?: string
 ): Promise<ShowStyleVariantId> {
-	return ShowStyleVariants.insertAsync({
+	await assertShowStyleBaseAccess(context, showStyleBaseId)
+
+	const highestRank = ShowStyleVariants.find(
+		{},
+		{
+			sort: {
+				_rank: -1,
+			},
+			limit: 1,
+		}
+	).fetch()[0]?._rank
+	const rank = highestRank !== undefined ? highestRank + 1 : 0
+
+	const showStyleVariant: ShowStyleVariant = {
 		_id: getRandomId(),
-		showStyleBaseId: showStyleBase._id,
 		name: name || 'Variant',
 		blueprintConfigWithOverrides: wrapDefaultObject({}),
 		_rundownVersionHash: '',
-	})
+		_rank: rank,
+		showStyleBaseId: showStyleBaseId,
+	}
+
+	return insertShowStyleVariant(showStyleVariant)
+}
+
+export async function importShowStyleVariant(
+	context: MethodContext | Credentials,
+	showStyleVariant: ShowStyleVariant
+): Promise<ShowStyleVariantId> {
+	await assertShowStyleBaseAccess(context, showStyleVariant.showStyleBaseId)
+
+	return insertShowStyleVariant(showStyleVariant)
 }
+
+export async function importShowStyleVariantAsNew(
+	context: MethodContext | Credentials,
+	showStyleVariant: ShowStyleVariant
+): Promise<ShowStyleVariantId> {
+	await assertShowStyleBaseAccess(context, showStyleVariant.showStyleBaseId)
+
+	showStyleVariant._id = getRandomId()
+
+	return insertShowStyleVariant(showStyleVariant)
+}
+
+async function insertShowStyleVariant(showStyleVariant: ShowStyleVariant): Promise<ShowStyleVariantId> {
+	return ShowStyleVariants.insertAsync(showStyleVariant)
+}
+
 export async function removeShowStyleBase(context: MethodContext, showStyleBaseId: ShowStyleBaseId): Promise<void> {
-	check(showStyleBaseId, String)
-	const access = await ShowStyleContentWriteAccess.anyContent(context, showStyleBaseId)
-	const showStyleBase = access.showStyleBase
-	if (!showStyleBase) throw new Meteor.Error(404, `showStyleBase "${showStyleBaseId}" not found`)
+	await assertShowStyleBaseAccess(context, showStyleBaseId)
 
 	await Promise.allSettled([
-		ShowStyleBases.removeAsync(showStyleBase._id),
+		ShowStyleBases.removeAsync(showStyleBaseId),
 		ShowStyleVariants.removeAsync({
-			showStyleBaseId: showStyleBase._id,
+			showStyleBaseId: showStyleBaseId,
 		}),
 		RundownLayouts.removeAsync({
-			showStyleBaseId: showStyleBase._id,
+			showStyleBaseId: showStyleBaseId,
```

Comment:
minor note: `{ showStyleBaseId: showStyleBaseId }` can be shortened to `{ showStyleBaseId }`